docs/build/tools/foundry.mdx

```diff
@@ -0,0 +1,246 @@
+---
+description: A guide for Foundry
+---
+
+# Foundry
+
+[Foundry] is a smart contract development environment
+for EVM-based chains. This guide will show you how to use Foundry to build,
+test, and deploy smart contracts on Oasis Sapphire.
+
+:::info
+
+For comprehensive details about Foundry's features, consult the
+[Foundry documentation].
+
+:::
+
+Foundry contains the following tools:
+
+- **Forge** is a CLI [tool] for building, testing,
+and deploying smart contracts.
+Due to integrated revm it significantly increases runtime speed when testing.
+
+- **Anvil** is a local development EVM [node].
+It is installed as part of Foundry, but currently cannot
+be extended with Sapphire features.
+
+- **Cast** is a [CLI tool] for interacting with EVM nodes.
+It uses RPC calls, so it can be used
+to interact with Sapphire nodes running the [Oasis Web3 gateway].
+
+- **Chisel** is a Solidity [REPL] (short for "read-eval-print loop")
+that allows developers to write and test Solidity code snippets.
+It provides an interactive environment for writing and executing
+Solidity code, as well as a set of built-in commands for working
+with and debugging your code.
+
+
+## Setup and Configuration
+
+Foundry uses default revm EVM implementation which does not contain
+[Sapphire-specific features].
+For that reason, we have to install special Sapphire precompiles which are
+available in the **Sapphire foundry** package.
+
+**Sapphire Contracts** package is also provided. It contains
+helper solidity contracts and libraries that enable convenient access to
+on-chain data and precompiles (rng, signatures, on-chain encryption, etc.).
+
+Follow the steps below to setup your project:
+
+1. Install and run Foundryup:
+
+   ```shell
+   curl -L https://foundry.paradigm.xyz | bash
+   foundryup
+   source ~/.bashrc
+   ```
+
+2. Create a new Forge project and move inside:
+
+   ```shell
+   forge init sapphire_demo
+   cd sapphire_demo
+   ```
+
+3. After intializing the project, we can install the **Sapphire Contracts**
+package using Foundry package manager [soldeer]:
+
+   ```shell
+   forge soldeer install @oasisprotocol-sapphire-contracts~0.2.11
```

Comment:
The version could be omitted right?

```suggestion
   forge soldeer install @oasisprotocol-sapphire-contracts
```

Reply:
I couldn't find a way to make it work without specifying version. The suggested change returns an error, and in soldeer/foundry docs, version is always mentioned.

Reply:
Since these are step-by-step instructions I think it's better to leave it as is. `forge soldeer install` shortcut and foundry.toml are already mentioned in the later section.